src/components/common/App/AppInput.vue

```diff
@@ -9,28 +12,119 @@ export default defineComponent({
     modelValue: {
       type: String,
       required: true,
+    },
+    label: {
+      type: String,
+    },
+    type: {
+      type: String,
+      default: 'text'
+    },
+    isValid: {
+      type: Boolean,
+      default: true,
+    },
+  },
+  data() {
+    return {
+      eyeClicked: false,
     }
   },
-  emits: ['update:modelValue'],
+  emits: ['update:modelValue', 'blur'],
+  methods: {
+    toggleVisibility() {
+      const inputSibling = document.getElementById('password');
```

Comment:
That's not how we should do this 👀
If we need to change that element's type - have a data property for that, bind it to the element and change it instead, DOM operations for such things aren't a good practice 😄 

If in any other case we need to do something more directly with the template (usually we really don't 😅) then we should use `template refs`: https://vuejs.org/guide/essentials/template-refs.html

Reply:
Ahh, this is so simple! ✨ I was convinced that my choice is either to manipulate with DOM or to overwrite the 'type' prop, so I chose the first option, although wasn't happy with that solution.